index.js

```diff
@@ -1,21 +1,542 @@
+    var projectParameter;
+
+    //download testing framework
+    await downloadLatestFramework(libraryVersion);
+
+    if (workspace) {
+      console.log(`Workspace selected: ${workspace}`);
+      projectParameter = "-workspace " + `"${workspace}"`;
+    } else if (xcodeproj) {
+      console.log(`Project selected: ${xcodeproj}`);
+      projectParameter = "-project " + `"${xcodeproj}"`;
+    } else if (fs.existsSync("Package.swift")) {
+      console.log(`Package.swift selected`);
+      await swiftPackageRun(platform, extraParameters);
+      return;
+    } else {
+      core.setFailed(
+        "Unable to find workspace, project or Swift package file. Please set with workspace or xcodeproj"
+      );
+    }
+
+    const scheme = await getScheme(workspace, xcodeproj);
+    console.log(`Scheme selected: ${scheme}`);
+
+    //copy configfile
+    const configfileName = "ddTesting.xcconfig";
+    const configFilePath = sdkTestingDir + "/" + configfileName;
+    createXCConfigFile(configFilePath, sdkTestingDir);
+
+
+
+    let codeCoverParam = "-enableCodeCoverage YES";
+
+    //build for testing
+    const buildCommand = `xcodebuild build-for-testing ${codeCoverParam} ` +
+    `-xcconfig ${configFilePath} ${projectParameter}` +
+    `-configuration ${configuration} ` +
+    `-scheme "${scheme}" `+
+    `-sdk ${sdk} ` +
+    `-derivedDataPath ${derivedDataPath} ` +
+    `-destination "${destination}"` +
+    extraParameters
+    const result = await exec.exec(buildCommand, null, null);
+
+    //For all testruns that are configured
+    let testRuns = await getXCTestRuns();
+    let testError;
+
+    for (const testRun of testRuns) {
+      //modify xctestrun with Scope variables
+
+      let plutilExportCommand =
+        "plutil -convert json -o " + testrunJson + ` "${testRun}"`;
+      await exec.exec(plutilExportCommand, null, null);
 
-    core.debug((new Date()).toTimeString()); // debug is only output if you set the secret `ACTIONS_RUNNER_DEBUG` to true
-    await wait(parseInt(ms));
-    core.info((new Date()).toTimeString());
+      let jsonString = fs.readFileSync(testrunJson, "utf8");
+      const testTargets = JSON.parse(jsonString);
+
+      for (const target of Object.keys(testTargets)) {
+        if (target.charAt(0) !== "_") {
+          if (testTargets[target].TestingEnvironmentVariables) {
+            await insertEnvVariables(testRun, target);
+          } else if (target === "TestConfigurations") {
+            let configurationNumber = 0;
+            for (const configuration of testTargets["TestConfigurations"]) {
+              let testNumber = 0;
+              for (const test of configuration["TestTargets"]) {
+                await insertEnvVariables(
+                  testRun,
+                  target +
+                    "." +
+                    configurationNumber +
+                    "." +
+                    "TestTargets" +
+                    "." +
+                    testNumber
```

Comment:
maybe this is easier to read? 
```suggestion
                await insertEnvVariables(
                  testRun,
                  `${target}.${configurationNumber}.TestTargets.${testNumber}`
```